build_stats_functions/build_event_protocol/build_event.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+
+"""Python Code for grabbing stats from build event protocol to be used."""
+import json
+import os
+import re
+from os import path
+
+import functions_framework
+from google.cloud import bigquery
+from google.cloud import logging 
+from google.cloud import storage
+
+class IncorrectFileTypeError(Exception):
+  """This file is not correctly formatted in JSON."""
+
+class EmptyFileError(Exception):
+  """The file passed in was empty"""
+
+class IncorrectFileFormatError(Exception):
+  """This file is not in the correct Build Event Protocol format."""
+
+
+# Triggered by a change in a storage bucket
+@functions_framework.cloud_event
+def main(cloud_event: any):
+  """Entry point function that is triggered by cloud event.
+  Gathers stats from Build Event Protocol file and sends to BigQuery.
+  Args:
+    cloud_event: event that triggers the function
+  """
+  STORAGE_BUCKET = os.environ.get("STORAGE_BUCKET")
+  TABLE_ID = os.environ.get("TABLE_ID")
+  try:
+    client = logging.Client()
+    log_name = "build-event-testing"
+    logger = client.logger(log_name)
+    logger.log_text("Successfully connected to GCP Logging Client", severity="INFO")
+  except logging.Error:
+    logger.log_text("Unable to connect to GCP Logging client", severity="WARNING")
+    raise logging.Error()
+  data = cloud_event.data
+  if "name" not in data:
+    logger.log_text("No filename was found", severity="WARNING")
+    return
+  file_name = data["name"]
+  check_file = check_path(file_name)
+  if not check_file:
+    logger.log_text(
+        "The current file is not a BEP according to its path", severity="INFO"
+    )
+    return
+  # Extract job name from overall path
+    logger.log_text("Build Event Protocol found", severity="INFO")
+    r = re.search("/[0-9].*", file_name)
+    index = r.span()[0]
+    job = file_name[0:index]
+  try:
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(STORAGE_BUCKET)
+    data_blob = bucket.get_blob(file_name)
+  except Exception as exc:
+    logger.log_text("Unable to access Cloud Storage client or storage bucket", severity="WARNING")
+    raise Exception() from exc
```

Comment:
Why not reraise the same exception if no additional value is being added to it?

Reply:
By additional value do you mean if there's not all_events (lines 82-84).